rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -575,6 +576,27 @@ bool microphone_device::has_data() const
 	return mic_registered && mic_opened && mic_started && (rbuf_raw.has_data() || rbuf_dsp.has_data());
 }
 
+f32 microphone_device::calculate_energy_level()
+{
+	const auto& buffer = device_type == microphone_handler::real_singstar ? ::at32(devices, 0).buf : temp_buf;
+
+	const be_t<s16>* samples = utils::bless<const be_t<s16>>(buffer.data());
```

Comment:
You can use read_from_ptr instead of pointer cast.

Reply:
Done. `convert_16_bit_pcm_to_float` may need to be cleaned up then as well, although I'm not sure if it should be done in this PR.